android-json-form-wizard/src/main/java/com/vijay/jsonwizard/customviews/TreeViewDialog.java

```diff
@@ -20,38 +20,49 @@
 import java.util.List;
 
 public class TreeViewDialog extends Dialog implements TreeNode.TreeNodeClickListener {
-    private static final String KEY_NODES = "nodes";
-    private static final String KEY_LEVEL = "level";
-    private static final String KEY_NAME = "name";
-    private static final String KEY_KEY = "key";
+    protected static final String KEY_NODES = "nodes";
+    protected static final String KEY_LEVEL = "level";
+    protected static final String KEY_NAME = "name";
+    protected static final String KEY_KEY = "key";
     private final Context context;
     private boolean shouldExpandAllNodes = false;
     private ArrayList<String> value;
     private ArrayList<String> name;
     private HashMap<TreeNode, String> treeNodeHashMap;
     private ArrayList<String> defaultValue;
+    private AndroidTreeView androidTreeView;
+    private LinearLayout canvas;
+    private boolean shouldDisableOnClickListener = false;
 
-    public TreeViewDialog(Context context, JSONArray structure, ArrayList<String> defaultValue, ArrayList<String> value)
+    public TreeViewDialog(Context context, JSONArray structure, ArrayList<String> defaultValue,
+                          ArrayList<String> value)
             throws
             JSONException {
         super(context);
         this.context = context;
-        init(structure, defaultValue, value);
+        init(structure, defaultValue, value, false);
+    }
+
+    public TreeViewDialog(Context context, int theme, JSONArray structure, ArrayList<String> defaultValue,
+                          ArrayList<String> value, boolean isSelectionMode) throws JSONException {
+        super(context, theme);
+        this.context = context;
+        init(structure, defaultValue, value, isSelectionMode);
     }
 
     public TreeViewDialog(Context context, int theme, JSONArray structure, ArrayList<String> defaultValue,
                           ArrayList<String> value) throws JSONException {
         super(context, theme);
         this.context = context;
-        init(structure, defaultValue, value);
+        init(structure, defaultValue, value, false);
```

Comment:
Why is the last parameter false by default? Does it mean it can never be `true` at any given chance?

Reply:
Yeah for the default implementation its false but for implemtations that need it to be true can this constructor https://github.com/opensrp/opensrp-client-native-form/blob/66a1d6a4c96546ac1e0cedf37bff85f7c06c0cb9/android-json-form-wizard/src/main/java/com/vijay/jsonwizard/customviews/TreeViewDialog.java#L46.